atomic_reactor/plugins/post_import_image.py

```diff
@@ -88,19 +88,28 @@ def run(self):
                           self.docker_image_repo)
 
             # Tags are imported automatically on creation
-            osbs.create_image_stream(self.imagestream, self.docker_image_repo,
-                                     **kwargs)
-        else:
-            self.log.info("Importing new tags for %s", self.imagestream)
-
-            attempts = 0
-            while not osbs.import_image(self.imagestream):
-                attempts += 1
-
-                if attempts >= self.import_attempts:
-                    msg = "Failed to import new tags for %s"
-                    raise RuntimeError(msg % self.imagestream)
-
-                self.log.info("no new tags, will retry after %d seconds (%d/%d)",
-                              self.retry_delay, attempts, self.import_attempts)
-                sleep(self.retry_delay)
+            i = osbs.create_image_stream(self.imagestream, self.docker_image_repo,
+                                         **kwargs)
+        self.log.info("Importing new tags for %s", self.imagestream)
+
+        primaries = None
+        try:
+            primaries = self.workflow.build_result.annotations['repositories']['primary']
+        except:
```

Comment:
Catching all exceptions is dangerous (e.g. if the build is canceled, the exception would be thrown - and it would be ignored).

Should be `except KeyError` most likely